pyof/foundation/basic_types.py

```diff
@@ -421,10 +421,23 @@ def __init__(self, value=b''):  # noqa
         Raises:
             ValueError: If given value is not bytes.
         """
-        if not isinstance(value, bytes):
-            raise ValueError('BinaryData must contain bytes.')
+        value = self._pack_if_necessary(value)
         super().__init__(value)
 
+    @staticmethod
+    def _pack_if_necessary(value):
+        if hasattr(value, 'pack') and callable(value.pack):
+            value = value.pack()
+        elif not value:
```

Comment:
I wrote this not for the case where value is None, but for the cases where it is an empty typle or list for example, which is a behavior it already had. I do not think it is a good idea, but kept it simply because it was there, and people were using it.
I'll remove it.